airflow/providers/common/io/xcom/__init__.py

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import packaging.version
+
+__all__ = ["__version__"]
+
+__version__ = "1.3.0"
+
+
+try:
+    from airflow import __version__ as airflow_version
+except ImportError:
+    from airflow.version import version as airflow_version
+
+if packaging.version.parse(packaging.version.parse(airflow_version).base_version) < packaging.version.parse(
+    "2.9.0"
+):
+    raise RuntimeError(
+        f"The package `apache-airflow-providers-common-io:{__version__}` needs Apache Airflow 2.9.0+"
```

Comment:
Here we do not require airflow 2.9.0 for the whole provider, just for XCom Feature of it . so the mesage should be

"The Xcom backend featuer in common.io provider needs ...."

Reply:
Seems that the only reason why we need >= 2.9.0 is the new `purge` method.

Suggestion. Maybe we can really avoid that (and remove this file altogether. 

It looks like the only compatibility problem is lack of `purge` implementation in delete.  But for Airflow <2.9.0 we can easily override the whole delete method  (until we bump min airflow version to 2.9.0).

How about we (until min airflow version 2.9.0 for the whole provider) override delete method with a copy of the one from `arflow` - that would make it possible to use common.io.xcom.backend even in Airflow 2.6 

We could easily just implement delete method in `XComObjectStoreBackend` for example:

```python
    # TODO: Remove this method when we bump min_airflow_version to 2.9.0
    @classmethod
    @provide_session
    def delete(cls, xcoms: XCom | Iterable[XCom], session: Session) -> None:
        """Delete one or multiple XCom entries."""
       if ( if packaging.version.parse(packaging.version.parse(airflow_version).base_version) >=packaging.version.parse(
    "2.9.0") 
            return super.delete(xcoms=xcoms, session=session)
       if isinstance(xcoms, XCom):
            xcoms = [xcoms]
        for xcom in xcoms:
            if not isinstance(xcom, XCom):
                raise TypeError(f"Expected XCom; received {xcom.__class__.__name__}")
            XCom.purge(xcom, session)
            session.delete(xcom)
        session.commit()
```

I **think** that would work and that would show the REAL power and reason why want to move the XCom backend to provider - because it would give the provider to Airflow 2.8 users.

Reply:
Ah no, also clear method would have to be copied. Probably not worh it.

Reply:
Likely to prone to errors/easy to break.

Reply:
> Here we do not require airflow 2.9.0 for the whole provider, just for XCom Feature of it . so the mesage should be

it's just applicable to `common.io.xcom` package, not the whole provider.

However, @ephraimbuddy the core change is a bugfix, will it be possible to add it to 2.8.2? if so, we need to update the minimum version in this module.

Reply:
Haha I tried te be careful on the semantic versioning ;-). Indeed 'purge' is used at multiple places. One of the reasons I'd like to cleanup BaseXCom. We can go both ways here, 2.8.2 and treat it as a bug fix 'missing API component' or a change of API and make it 2.9.0.

Let me know: @ephraimbuddy what you prefer.

Reply:
I like @hussein-awala 's way of thinking here :)

Reply:
Would it be possible to make it two PRs, separating the core changes from the provider change? That way, it would be easier for users to know that it's a bug fix by merely reading the title.

Reply:
While possible, I'm not sure if that makes sense (and I am not looking forward to the work involved tbh). No one can make use of the 'bug fix'. I.e. it is a missing feature of the API and it will require a code change on the side of custom implementations to make use of it. It is a bug fix in the sense that you would expect the API to have it as it requires manual clean up on the side of the custom implementation now. 

In other words I'd like to keep it combined and just decide if the whole thing targets 2.9.0 or 2.8.2.

Reply:
Ok. I'm okay with whichever one. I will remember it if marked 2.8.2

Reply:
Yeah. Better to keep those in one PR. I think - contrary to  natural thinking - this is precisely the thing that is better when you have monorepo - that you **can** introduce changes to  multiple independent parts at the same time, and while resulting setup is more complex (because you have to manage back-compatibility and dependencies as we do here) generally development of it is quite a bit easier, because you effectively work on a combined change even if your change comes to two different components withing the monorepo.

And it's not a problem at all in our release process either - there is completely separate release process and marking "PR" as being part either of the provider or or airflow release - the same PR will appear in both at some point of time and we can literally decide at the last moment before relasing provider whethere we mark the change also cherry-pickable to 2.8 branch.

This is the TRUE power of monorepo - development of such combined features is WAY easier.

Reply:
So basically - as we use monorepo, there is virtually no benefit of splitting such PRs (and no need either).

Reply:
Shall we change it to 2.8.2 then @bolkedebruin ?

Reply:
Ill target 2.8.2